src/main/java/com/stuypulse/robot/subsystems/Shooter.java

```diff
@@ -0,0 +1,120 @@
+package com.stuypulse.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.stuypulse.robot.Constants;
+import com.stuypulse.robot.util.NEOEncoder;
+
+import edu.wpi.first.wpilibj.controller.PIDController;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Shooter extends SubsystemBase {
+
+    // Shooter Motors
+    // note: engineering wants to change to 2 motors to free up a spot on the pdp
+    private CANSparkMax leftShooterMotor;
+    private CANSparkMax rightShooterMotor;
+    private CANSparkMax middleShooterMotor;
+
+    // Shooter Encoders
+    private NEOEncoder leftShooterEncoder;
+    private NEOEncoder rightShooterEncoder; 
+    private NEOEncoder middleShooterEncoder;
+
+    // Feeder Motor
+    private CANSparkMax feederMotor;
+
+    // Hood Solenoid
+    private Solenoid hoodSolenoid;
+
+    // SpeedControllerGroup
+    private SpeedControllerGroup shooterMotors; 
+
+    // PIDController
+    private PIDController shooterController;
+
+    public Shooter() {
+        leftShooterMotor = new CANSparkMax(Constants.LEFT_SHOOTER_MOTOR_PORT, MotorType.kBrushless);
+        rightShooterMotor = new CANSparkMax(Constants.RIGHT_SHOOTER_MOTOR_PORT, MotorType.kBrushless);
+        middleShooterMotor = new CANSparkMax(Constants.MIDDLE_SHOOTER_MOTOR_PORT, MotorType.kBrushless);
+
+        leftShooterMotor.setInverted(true); //engineering said that the left motor is inverted
+
+        leftShooterEncoder = new NEOEncoder(leftShooterMotor.getEncoder());
+        rightShooterEncoder = new NEOEncoder(rightShooterMotor.getEncoder());
+        middleShooterEncoder = new NEOEncoder(middleShooterMotor.getEncoder());
+
+        feederMotor = new CANSparkMax(Constants.FEEDER_MOTOR_PORT, MotorType.kBrushless);
+
+        hoodSolenoid = new Solenoid(Constants.HOOD_SOLENOID_PORT);
+
+        shooterMotors = new SpeedControllerGroup(leftShooterMotor, rightShooterMotor, middleShooterMotor);
+
+        shooterController = new PIDController(Constants.SHOOTER_SHOOT_KP, Constants.SHOOTER_SHOOT_KI, Constants.SHOOTER_SHOOT_KD);
+    }
+
+    public double getError(double targetVelocity) {
+        return targetVelocity - getCurrentShooterVelocityInRPM();
+        // the erorr is calculated by subtracting the target velocity by the current velocity.
+    }
+    
+    public double getRawMaxShooterVelocity() {
```

Comment:
this isn't clear, Max makes it sound like you're getting the maximum possible velocity